File: Caronae/CaronaeConstants.m
#import "CaronaeConstants.h"

#pragma mark - Static pages URLs

NSString *const CaronaeIntranetURLString = @"https://api.caronae.com.br/login";
NSString *const CaronaeAboutPageURLString = @"https://caronae.com.br/sobre_mobile.html";
NSString *const CaronaeTermsOfUsePageURLString = @"https://caronae.com.br/termos_mobile.html";
NSString *const CaronaeFAQPageURLString = @"https://caronae.com.br/faq.html?mobile";


#pragma mark - Notifications

NSNotificationName const CaronaeNotificationReceivedNotification = @"CaronaeNotificationReceivedNotification";
NSNotificationName const CaronaeDidUpdateNotifications = @"CaronaeDidUpdateNotifications";
NSNotificationName const CaronaeDidUpdateUserNotification = @"CaronaeDidUpdateUserNotification";


#pragma mark - Preference keys

NSString *const CaronaePreferenceLastSearchedNeighborhoodsKey = @"lastSearchedNeighborhoods";
NSString *const CaronaePreferenceLastSearchedZoneKey = @"lastSearchedZone";
NSString *const CaronaePreferenceLastSearchedCampusKey = @"lastSearchedCampus";
NSString *const CaronaePreferenceLastSearchedCentersKey = @"lastSearchedCenters";
NSString *const CaronaePreferenceLastSearchedDateKey = @"lastSearchedDate";

NSString *const CaronaePreferenceFilterIsEnabledKey = @"filterIsEnabled";
NSString *const CaronaePreferenceLastFilteredZoneKey = @"lastFilteredZone";
NSString *const CaronaePreferenceLastFilteredNeighborhoodsKey = @"lastFilteredNeighborhoods";
NSString *const CaronaePreferenceLastFilteredCampusKey = @"lastFilteredCampus";
NSString *const CaronaePreferenceLastFilteredCentersKey = @"lastFilteredCenters";


#pragma mark - Etc.

NSString *const CaronaeErrorDomain = @"br.ufrj.caronae.error";
NSString *const CaronaeSignOutRequiredKey = @"CaronaeSignOutRequired";
NSString *const Caronae8PhoneNumberPattern = @"(###) ####-####";
NSString *const Caronae9PhoneNumberPattern = @"(###) #####-####";
NSString *const CaronaePlaceholderProfileImage = @"Profile Picture";
NSString *const CaronaeSearchDateFormat = @"EEEE, dd/MM/yyyy HH:mm";
NSString *const CaronaeDateLocaleIdentifier = @"pt_BR";
NSString *const CaronaeAllNeighborhoodsText = @"Todos os Bairros";
NSString *const CaronaeAllCampiText = @"Todos os Campi";
NSString *const CaronaeOtherZoneText = @"Outra";
NSString *const CaronaeOtherNeighborhoodsText = @"Outros";


@interface CaronaeConstants ()
@property (nonatomic, readwrite) UIColor *otherZoneColor;
@end

@implementation CaronaeConstants

+ (instancetype)defaults {
    static CaronaeConstants *sharedMyManager = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedMyManager = [[self alloc] init];
    });
    return sharedMyManager;
}

- (UIColor *) otherZoneColor {
    if (!_otherZoneColor) {
        _otherZoneColor = [UIColor colorWithWhite:0.541 alpha:1.000];
    }
    return _otherZoneColor;
}

@end

File: Caronae/CaronaeConstants.swift
import Foundation
// swiftlint:disable identifier_name


// MARK: Caronae API Base URL

#if DEVELOPMENT
    let CaronaeAPIBaseURLString = "https://api.dev.caronae.org"
#elseif TESTING
    let CaronaeAPIBaseURLString = "https://private-443d4-caronae.apiary-mock.com"
#else
    let CaronaeAPIBaseURLString = "https://api.caronae.org"
#endif


// MARK: Static pages URLs

struct CaronaeURLString {
    #if DEVELOPMENT
    static let base = "https://dev.caronae.org"
    #else
    static let base = "https://caronae.org"
    #endif
    static let login = String(format: "%@/login?type=app_jwt", CaronaeAPIBaseURLString)
    static let loginCallback = "caronae://login"
    static let aboutPage = String(format: "%@/sobre_mobile.html", CaronaeURLString.base)
    static let termsOfUsePage = String(format: "%@/termos_mobile.html", CaronaeURLString.base)
    static let FAQPage = String(format: "%@/faq.html?mobile", CaronaeURLString.base)
}


// MARK: Notifications

extension Foundation.Notification.Name {
    static let CaronaeDidUpdateNotifications = Foundation.Notification.Name("CaronaeDidUpdateNotifications")
    static let CaronaeDidUpdateUser = Foundation.Notification.Name("CaronaeDidUpdateUserNotification")
}

// MARK: Preference keys

struct CaronaePreferenceLastSearch {
    static let zoneKey = "lastSearchedZone"
    static let neighborhoodsKey = "lastSearchedNeighborhoods"
    static let campusKey = "lastSearchedCampus"
    static let centersKey = "lastSearchedCenters"
    static let dateKey = "lastSearchedDate"
}

struct CaronaePreferenceLastOfferedRide {
    static let key = "lastOfferedRide"
    static let zone = "zone"
    static let neighborhood = "neighborhood"
    static let place = "place"
    static let route = "route"
    static let slots = "slots"
    static let description = "description"
    static let hubGoing = "hubGoing"
    static let hubReturning = "hubReturning"
}

struct CaronaePreferenceLastFilter {
    static let isEnabledKey = "filterIsEnabled"
    static let zoneKey = "lastFilteredZone"
    static let neighborhoodsKey = "lastFilteredNeighborhoods"
    static let campusKey = "lastFilteredCampus"
    static let centersKey = "lastFilteredCenters"
}


// MARK: Etc.

let CaronaeErrorDomain = "br.ufrj.caronae.error"
let CaronaeSignOutRequiredKey = "CaronaeSignOutRequired"
let Caronae9PhoneNumberPattern = "({0}[00]) [00000]-[0000]"
let CaronaeBrazilianCarPlatePattern = "[AAA]{-}[0000]"
let CaronaePlaceholderProfileImage = "Profile Picture"
let CaronaeSearchDateFormat = "EEEE, dd/MM/yyyy HH:mm"
let CaronaeAllNeighborhoodsText = "Todos os Bairros"
let CaronaeAllCampiText = "Todos os Campi"
let CaronaeOtherNeighborhoodsText = "Outros"

let OtherZoneColor = UIColor(white: 0.541, alpha: 1.0)
